rust/src/go/nmstate/nmstate.go

```diff
@@ -236,3 +236,46 @@ func (n *Nmstate) GenerateConfiguration(state string) (string, error) {
 	}
 	return C.GoString(config), nil
 }
+
+// GenerateConfigurationFromPolicy generates the configuration from the provided policy and current state.
+// This function returns the configuration files based on the policy and current state provided.
+func (n *Nmstate) GenerateConfigurationFromPolicy(policy, currentState string) (string, error) {
```

Comment:
the "GenerateConfig" comes from the "gc" command at nmstatectl that generate NetworkManager configuration, but this is generate a NMstate witch is different, 

```suggestion
func (n *Nmstate) GenerateStateFromPolicy(policy, currentState string) (string, error) {
```

Reply:
I updated it, thanks!